h2/src/main/org/h2/pagestore/PageStore.java

```diff
@@ -477,7 +477,8 @@ public synchronized void compact(int compactMode) {
             return;
         }
         if (SysProperties.MODIFY_ON_WRITE && readMode &&
-                compactMode == 0) {
+                compactMode == 0 ||
+                 compactMode == CommandInterface.SHUTDOWN_IMMEDIATELY) {
```

Comment:
Such line wrapping looks weird, it's better to place `compactMode == 0` on the same line with `&&` conditions to group all operands of logical AND together and have logical OR operands on different lines.

Indentation of line 481 has one extra space character, build script will complain about it.